Source/DafnyCore/Rewriters/RefinementTransformer.cs

```diff
@@ -596,51 +596,76 @@ public RefinementTransformer(Program p)
       }
     }
 
-    Method CloneMethod(Method m, List<AttributedExpression> moreEnsures, Specification<Expression> decreases, BlockStmt newBody, bool checkPreviousPostconditions, Attributes moreAttributes) {
+    Method CloneMethod(Method m, Method newDeclaration, Specification<Expression> decreases, BlockStmt newBody, bool checkPreviousPostconditions, Attributes moreAttributes) {
       Contract.Requires(m != null);
       Contract.Requires(!(m is Constructor) || newBody == null || newBody is DividedBlockStmt);
       Contract.Requires(decreases != null);
+      List<AttributedExpression> moreEnsures = newDeclaration.Ens;
+      IToken newMethodTok = new NestedToken(newDeclaration.tok, m.tok, "it refines the following abstract declaration");
```

Comment:
Here I just indented the code and wrapped it with the call `refinementCloner.WithNewDeclarationTokenWrapping`
and I changed the use of "tok" with the `newMethodTok` above